cirq/devices/grid_qubit.py

```diff
@@ -38,6 +38,61 @@ def is_adjacent(self, other: ops.Qid) -> bool:
         return (isinstance(other, GridQubit) and
                 abs(self.row - other.row) + abs(self.col - other.col) == 1)
 
+    @staticmethod
+    def square(size: int, top: int = 0, left: int = 0) -> List['GridQubit']:
+        """Returns a square of GridQubits
+
+        Args:
+            size: Length of a side of the square
+            top: Row number of the topmost row
+            left: Column number of the leftmost row
+
+        Returns:
+            A list of GridQubits filling in a square grid
+        """
+        return [GridQubit(row, col)
+                for row in range(top, top+size)
+                for col in range(left, left+size)]
+
+    @staticmethod
+    def rect(rows: int, cols: int,
+             top: int = 0, left: int = 0) -> List['GridQubit']:
+        """Returns a rectangle of GridQubits
+
+        Args:
+            rows: Number of rows in the rectangle
+            cols: Number of columns in the rectangle
+            top: Row number of the topmost row
+            left: Column number of the leftmost row
+
+        Returns:
+            A list of GridQubits filling in a rectangular grid
+        """
+        return [GridQubit(row, col)
+                for row in range(top, top+rows)
+                for col in range(left, left+cols)]
+
+    @staticmethod
+    def from_pic(s: str) -> List['GridQubit']:
+        """Parse ASCIIart device layout into info about qubits and connectivity.
+
+        Args:
+            s: String representing the qubit layout. Each line represents a row,
+                and each character in the row is a qubit, or a blank site if the
+                character is a hyphen '-'.
+
+        Returns:
+            A list of GridQubits corresponding to the provided picture
+        """
+        lines = s.strip().split('\n')
+        qubits = []
+        for row, line in enumerate(lines):
+            for col, c in enumerate(line.strip()):
+                if c != '-':
```

Comment:
Don't just test for not hyphen, also define an acceptable character set for "yes qubit here".

Shouldn't the not-a-qubit character be space?

Reply:
I was following the same convention used in for how to translate the diagrams. It seemed intuitive enough to just translate the parts of this private method that do the desired task over to GridQubit. Thoughts?
https://github.com/quantumlib/Cirq/blob/0452fd817fd6a52e23682712484a8a5c047d30d0/cirq/google/known_devices.py#L22-L31

Reply:
I have sort of a "whatever you do in the privacy of your own module is your business" attitude to these things. When the method was private I didn't worry about which characters were used. But if we're exposing it then that question is suddenly very relevant.

I would say we should allow the use of dot, dash, and space as no-qubit characters and limit the qubit characters to be alphanumeric.

Reply:
Ok, I'll implement that and update the documentation.